mantle/cosa/schema.go

```diff
@@ -0,0 +1,78 @@
+package cosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+
+	schema "github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	// JSON Schema document. Default the generated Schema.
+	SchemaJSON = generatedSchemaJSON
+)
+
+func init() {
+	runtimeSchemaPath := os.Getenv("COSA_META_SCHEMA")
+	if runtimeSchemaPath != "" {
+		f, err := os.Open(runtimeSchemaPath)
+		if err != nil {
+			panic(errors.Wrapf(err, "failed to open schema file %s", runtimeSchemaPath))
+		}
+		defer f.Close()
+
+		if err := SetSchemaFromFile(f); err != nil {
+			panic(errors.Wrapf(err, "failed to read in schema file %s", runtimeSchemaPath))
+		}
+	}
+}
+
+// SetSchemaFromFile sets the validation JSON Schema
+func SetSchemaFromFile(r io.Reader) error {
+	if r == nil {
+		return errors.New("schema input is invalid")
+	}
+	in, err := ioutil.ReadAll(r)
+	if err != nil {
+		SchemaJSON = string(in)
+	}
+	return nil
+}
+
+// Validator checks the build against the schema.
+func (build *Build) Validator() (bool, []error) {
```

Comment:
Is there ever a case where a config can be valid and have no errors? If not then it's probably safe to drop the boolean and when checking if it's a valid config just do something like:
```golang
if err := build.Validator(); err != nil {
    return fmt.Errorf("error: %v", err)
}
// build is valid
```

Reply:
Fixed up.